pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2244,9 +2244,18 @@ private CompletableFuture<Integer> unloadServiceUnit(NamespaceBundle serviceUnit
                 closeFutures.add(topicFuture
                         .thenCompose(t -> t.isPresent() ? t.get().close(
                                 disconnectClients, closeWithoutWaitingClientDisconnect)
-                                : CompletableFuture.completedFuture(null)));
+                                : CompletableFuture.completedFuture(null))
+                        .exceptionally(e -> {
+                            if (e.getCause() instanceof BrokerServiceException.ServiceUnitNotReadyException
+                                    && e.getMessage().contains("Please redo the lookup")) {
+                                log.warn("[{}] Topic ownership check failed. Skipping it", topicName);
+                                return null;
```

Comment:
Do we have a test to cover this change? I noticed that even after removing this change, the `testUnloadUponTopicLookupFailure` test still passes.

Reply:
Interesting. I see the test fails if I remove this on my laptop.
<img width="1412" alt="Screen Shot 2024-03-28 at 7 35 45 PM" src="https://github.com/apache/pulsar/assets/103456639/ea53d7a6-4261-4654-93cb-0eef0b59e2fc">

Reply:
Can confirm @Demogorgon314's finding. The test doesn't always execute the respective codepath.

Reply:
You also need to update configs like in this PR. Otherwise, the monitor job will clean the stuck unloading state.

Reply:
Identified another racing condition here. 
UnloadManager should not complete the inflight jobs from the Owned event from assignment commands. This could result in returning unload command too fast, although the unloading is still in progress. I updated this fix.

Reply:
Thanks, this works now!

Reply:
Good catch!